{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     PATIENT_ID                      OTHER_PATIENT_ID  \\\n",
      "0  TCGA-3X-AAV9  41b97b11-acaa-4fbc-b3b0-0abc1bcac13b   \n",
      "1  TCGA-3X-AAVA  fe57b639-db7a-460f-adfe-552f1e034e46   \n",
      "2  TCGA-3X-AAVB  3824cd6d-c85c-4b21-819a-932d1afef976   \n",
      "3  TCGA-3X-AAVC  ff4131c9-537c-46cc-b495-9af60d431f5e   \n",
      "4  TCGA-3X-AAVE  dcc87c59-fd8c-4511-9983-96d5b9784cb0   \n",
      "\n",
      "                PRIMARY_SITE_PATIENT                  DISEASE_TYPE  \\\n",
      "0  Liver and intrahepatic bile ducts  Adenomas and Adenocarcinomas   \n",
      "1  Liver and intrahepatic bile ducts  Adenomas and Adenocarcinomas   \n",
      "2  Liver and intrahepatic bile ducts  Adenomas and Adenocarcinomas   \n",
      "3                        Gallbladder  Adenomas and Adenocarcinomas   \n",
      "4  Liver and intrahepatic bile ducts  Adenomas and Adenocarcinomas   \n",
      "\n",
      "         PROJECT_NAME PROJECT_ID     SEX   RACE               ETHNICITY  \\\n",
      "0  Cholangiocarcinoma  TCGA-CHOL    Male  ASIAN  NOT HISPANIC OR LATINO   \n",
      "1  Cholangiocarcinoma  TCGA-CHOL  Female  WHITE  NOT HISPANIC OR LATINO   \n",
      "2  Cholangiocarcinoma  TCGA-CHOL  Female  WHITE  NOT HISPANIC OR LATINO   \n",
      "3  Cholangiocarcinoma  TCGA-CHOL  Female  WHITE  NOT HISPANIC OR LATINO   \n",
      "4  Cholangiocarcinoma  TCGA-CHOL    Male  ASIAN  NOT HISPANIC OR LATINO   \n",
      "\n",
      "  VITAL_STATUS  ...  MORPHOLOGY PATH_T_STAGE  PRIOR_TREATMENT PATH_N_STAGE  \\\n",
      "0         Dead  ...      8160/3           T1            False           N0   \n",
      "1         Dead  ...      8160/3          T2b            False           NX   \n",
      "2        Alive  ...      8160/3           T3            False           N1   \n",
      "3        Alive  ...      8160/3           T1            False           N0   \n",
      "4        Alive  ...      8160/3           T2            False           N0   \n",
      "\n",
      "  PRIOR_MALIGNANCY PROJECT_STATE   OS_STATUS  OS_MONTHS  \\\n",
      "0             True      released  1:DECEASED  11.136662   \n",
      "1            False      released  1:DECEASED  14.618922   \n",
      "2            False      released    0:LIVING  13.206307   \n",
      "3            False      released    0:LIVING  23.291721   \n",
      "4            False      released    0:LIVING  21.353482   \n",
      "\n",
      "              DFS_STATUS   DFS_MONTHS  \n",
      "0  1:Recurred/Progressed  7.101292041  \n",
      "1  1:Recurred/Progressed  8.679356939  \n",
      "2                    NaN          NaN  \n",
      "3          0:DiseaseFree   23.3093336  \n",
      "4          0:DiseaseFree  21.36962883  \n",
      "\n",
      "[5 rows x 29 columns]\n",
      "  Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
      "0       DOCK7           85440  WUGSC     GRCh38          1        62552814   \n",
      "1      LRRC71          149499  WUGSC     GRCh38          1       156932006   \n",
      "2      PTPN14            5784  WUGSC     GRCh38          1       214383971   \n",
      "3      OR2T12          127064  WUGSC     GRCh38          1       248295007   \n",
      "4      PARGP1          728407  WUGSC     GRCh38         10        45892481   \n",
      "\n",
      "   End_Position Strand                         Consequence  \\\n",
      "0      62552814      +                    missense_variant   \n",
      "1     156932006      +                    missense_variant   \n",
      "2     214383971      +                  synonymous_variant   \n",
      "3     248295007      +                    missense_variant   \n",
      "4      45892481      +  non_coding_transcript_exon_variant   \n",
      "\n",
      "  Variant_Classification  ...                         HGVSc        HGVSp  \\\n",
      "0      Missense_Mutation  ...   ENST00000635253.2:c.2684G>A  p.Arg895Gln   \n",
      "1      Missense_Mutation  ...   ENST00000337428.8:c.1420C>T  p.Leu474Phe   \n",
      "2                 Silent  ...  ENST00000366956.10:c.1884C>T    p.Leu628=   \n",
      "3      Missense_Mutation  ...    ENST00000641276.1:c.572C>T  p.Ser191Leu   \n",
      "4                    RNA  ...   ENST00000417555.2:n.2089A>C          NaN   \n",
      "\n",
      "  HGVSp_Short    Transcript_ID          RefSeq  Protein_position   Codons  \\\n",
      "0     p.R895Q  ENST00000635253  NM_001367561.1             895.0  cGa/cAa   \n",
      "1     p.L474F  ENST00000337428     NM_144702.3             474.0  Ctt/Ttt   \n",
      "2     p.L628=  ENST00000366956     NM_005401.5             628.0  ctC/ctT   \n",
      "3     p.S191L  ENST00000641276  NM_001004692.2             191.0  tCa/tTa   \n",
      "4     p.*697*  ENST00000417555             NaN               NaN      NaN   \n",
      "\n",
      "  Exon_Number  genomic_location_explanation  Annotation_Status  \n",
      "0       22/50                           NaN            SUCCESS  \n",
      "1       13/15                           NaN            SUCCESS  \n",
      "2       13/19                           NaN            SUCCESS  \n",
      "3         3/3                           NaN            SUCCESS  \n",
      "4       11/12                           NaN            SUCCESS  \n",
      "\n",
      "[5 rows x 47 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# Directory containing the data files\n",
    "directory = '/Users/ethan/Downloads/tcga_data'\n",
    "\n",
    "# Initialize dictionaries to store DataFrames for clinical and mutation files\n",
    "clinical_data = {}\n",
    "mutation_data = {}\n",
    "\n",
    "# Loop through each file in the directory\n",
    "for filename in os.listdir(directory):\n",
    "    cancer_type = filename.split('_')[0]\n",
    "    file_path = os.path.join(directory, filename)\n",
    "\n",
    "    try:\n",
    "        if filename.endswith('_clinical_patient.txt'):\n",
    "            # Read clinical patient file, skipping comment lines\n",
    "            clinical_data[cancer_type] = pd.read_csv(file_path, sep='\\t', comment='#', header=0, engine='python')\n",
    "        elif filename.endswith('_mutations.txt'):\n",
    "            if cancer_type == 'chol':\n",
    "                # Skip the first two metadata lines in the chol_mutations.txt file\n",
    "                mutation_data[cancer_type] = pd.read_csv(file_path, sep='\\t', skiprows=2, engine='python')\n",
    "            else:\n",
    "                # Read other mutations files normally\n",
    "                mutation_data[cancer_type] = pd.read_csv(file_path, sep='\\t', header=0, engine='python')\n",
    "    except pd.errors.ParserError as e:\n",
    "        print(f\"Error parsing {filename}: {e}\")\n",
    "\n",
    "# Example: Access the DataFrame for 'chol' cancer type\n",
    "chol_clinical_df = clinical_data.get('chol')\n",
    "chol_mutation_df = mutation_data.get('chol')\n",
    "\n",
    "# Display the first few rows to verify\n",
    "print(chol_clinical_df.head())\n",
    "print(chol_mutation_df.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['PATIENT_ID', 'OTHER_PATIENT_ID', 'PRIMARY_SITE_PATIENT',\n",
       "       'DISEASE_TYPE', 'PROJECT_NAME', 'PROJECT_ID', 'SEX', 'RACE',\n",
       "       'ETHNICITY', 'VITAL_STATUS', 'YEAR_OF_DEATH', 'PRIMARY_DIAGNOSIS',\n",
       "       'YEAR_OF_DIAGNOSIS', 'PATH_M_STAGE', 'BIOPSY_SITE',\n",
       "       'AJCC_STAGING_EDITION', 'ICD_10', 'AGE', 'PATH_STAGE', 'MORPHOLOGY',\n",
       "       'PATH_T_STAGE', 'PRIOR_TREATMENT', 'PATH_N_STAGE', 'PRIOR_MALIGNANCY',\n",
       "       'PROJECT_STATE', 'OS_STATUS', 'OS_MONTHS', 'DFS_STATUS', 'DFS_MONTHS'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data['chol'].columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       NaN\n",
       "1       NaN\n",
       "2       NaN\n",
       "3       NaN\n",
       "4       NaN\n",
       "         ..\n",
       "53772   NaN\n",
       "53773   NaN\n",
       "53774   NaN\n",
       "53775   NaN\n",
       "53776   NaN\n",
       "Name: t_ref_count, Length: 53777, dtype: float64"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_data.get('lihc')['t_ref_count']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       TCGA-3X-AAV9-01A\n",
       "1       TCGA-3X-AAV9-01A\n",
       "2       TCGA-3X-AAV9-01A\n",
       "3       TCGA-3X-AAV9-01A\n",
       "4       TCGA-3X-AAV9-01A\n",
       "              ...       \n",
       "3759    TCGA-ZU-A8S4-01A\n",
       "3760    TCGA-ZU-A8S4-01A\n",
       "3761    TCGA-ZU-A8S4-01A\n",
       "3762    TCGA-ZU-A8S4-01A\n",
       "3763    TCGA-ZU-A8S4-01A\n",
       "Name: Tumor_Sample_Barcode, Length: 3764, dtype: object"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_data['chol']['Tumor_Sample_Barcode']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     TCGA-3X-AAV9\n",
       "1     TCGA-3X-AAVA\n",
       "2     TCGA-3X-AAVB\n",
       "3     TCGA-3X-AAVC\n",
       "4     TCGA-3X-AAVE\n",
       "5     TCGA-4G-AAZF\n",
       "6     TCGA-4G-AAZG\n",
       "7     TCGA-4G-AAZO\n",
       "8     TCGA-4G-AAZR\n",
       "9     TCGA-4G-AAZT\n",
       "10    TCGA-5A-A8ZF\n",
       "11    TCGA-5A-A8ZG\n",
       "12    TCGA-W5-AA2G\n",
       "13    TCGA-W5-AA2H\n",
       "14    TCGA-W5-AA2I\n",
       "15    TCGA-W5-AA2J\n",
       "16    TCGA-W5-AA2K\n",
       "17    TCGA-W5-AA2M\n",
       "18    TCGA-W5-AA2O\n",
       "19    TCGA-W5-AA2Q\n",
       "20    TCGA-W5-AA2R\n",
       "21    TCGA-W5-AA2T\n",
       "22    TCGA-W5-AA2U\n",
       "23    TCGA-W5-AA2W\n",
       "24    TCGA-W5-AA2X\n",
       "25    TCGA-W5-AA2Z\n",
       "26    TCGA-W5-AA30\n",
       "27    TCGA-W5-AA31\n",
       "28    TCGA-W5-AA33\n",
       "29    TCGA-W5-AA34\n",
       "30    TCGA-W5-AA36\n",
       "31    TCGA-W5-AA38\n",
       "32    TCGA-W5-AA39\n",
       "33    TCGA-W6-AA0S\n",
       "34    TCGA-W6-AA0T\n",
       "35    TCGA-W7-A93N\n",
       "36    TCGA-W7-A93O\n",
       "37    TCGA-W7-A93P\n",
       "38    TCGA-WD-A7RX\n",
       "39    TCGA-YR-A95A\n",
       "40    TCGA-ZD-A8I3\n",
       "41    TCGA-ZH-A8Y1\n",
       "42    TCGA-ZH-A8Y2\n",
       "43    TCGA-ZH-A8Y3\n",
       "44    TCGA-ZH-A8Y4\n",
       "45    TCGA-ZH-A8Y5\n",
       "46    TCGA-ZH-A8Y6\n",
       "47    TCGA-ZH-A8Y7\n",
       "48    TCGA-ZH-A8Y8\n",
       "49    TCGA-ZK-AAYZ\n",
       "50    TCGA-ZU-A8S4\n",
       "Name: PATIENT_ID, dtype: object"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data['chol']['PATIENT_ID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Match_Norm_Seq_Allele1</th>\n",
       "      <th>Match_Norm_Seq_Allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOCK7</td>\n",
       "      <td>85440</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>62552814</td>\n",
       "      <td>62552814</td>\n",
       "      <td>1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LRRC71</td>\n",
       "      <td>149499</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>156932006</td>\n",
       "      <td>156932006</td>\n",
       "      <td>1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PTPN14</td>\n",
       "      <td>5784</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>214383971</td>\n",
       "      <td>214383971</td>\n",
       "      <td>1</td>\n",
       "      <td>Silent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR2T12</td>\n",
       "      <td>127064</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>248295007</td>\n",
       "      <td>248295007</td>\n",
       "      <td>1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PARGP1</td>\n",
       "      <td>728407</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>45892481</td>\n",
       "      <td>45892481</td>\n",
       "      <td>10</td>\n",
       "      <td>RNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3759</th>\n",
       "      <td>PI15</td>\n",
       "      <td>51050</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>74825486</td>\n",
       "      <td>74825486</td>\n",
       "      <td>8</td>\n",
       "      <td>Silent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3760</th>\n",
       "      <td>KIAA2026</td>\n",
       "      <td>158358</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>6007401</td>\n",
       "      <td>6007401</td>\n",
       "      <td>9</td>\n",
       "      <td>Silent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3761</th>\n",
       "      <td>RABGAP1</td>\n",
       "      <td>23637</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>122990131</td>\n",
       "      <td>122990131</td>\n",
       "      <td>9</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3762</th>\n",
       "      <td>RABGAP1</td>\n",
       "      <td>23637</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>122990132</td>\n",
       "      <td>122990132</td>\n",
       "      <td>9</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3763</th>\n",
       "      <td>USP9X</td>\n",
       "      <td>8239</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>41229287</td>\n",
       "      <td>41229287</td>\n",
       "      <td>X</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3764 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol  Entrez_Gene_Id Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "0          DOCK7           85440                 C                 T   \n",
       "1         LRRC71          149499                 C                 T   \n",
       "2         PTPN14            5784                 G                 A   \n",
       "3         OR2T12          127064                 G                 A   \n",
       "4         PARGP1          728407                 T                 G   \n",
       "...          ...             ...               ...               ...   \n",
       "3759        PI15           51050                 C                 A   \n",
       "3760    KIAA2026          158358                 G                 A   \n",
       "3761     RABGAP1           23637                 G                 C   \n",
       "3762     RABGAP1           23637                 C                 T   \n",
       "3763       USP9X            8239                 C                 T   \n",
       "\n",
       "     Reference_Allele Variant_Type  Start_Position  End_Position Chromosome  \\\n",
       "0                   C          SNP        62552814      62552814          1   \n",
       "1                   C          SNP       156932006     156932006          1   \n",
       "2                   G          SNP       214383971     214383971          1   \n",
       "3                   G          SNP       248295007     248295007          1   \n",
       "4                   T          SNP        45892481      45892481         10   \n",
       "...               ...          ...             ...           ...        ...   \n",
       "3759                C          SNP        74825486      74825486          8   \n",
       "3760                G          SNP         6007401       6007401          9   \n",
       "3761                G          SNP       122990131     122990131          9   \n",
       "3762                C          SNP       122990132     122990132          9   \n",
       "3763                C          SNP        41229287      41229287          X   \n",
       "\n",
       "     Variant_Classification  Match_Norm_Seq_Allele1  Match_Norm_Seq_Allele2  \n",
       "0         Missense_Mutation                     NaN                     NaN  \n",
       "1         Missense_Mutation                     NaN                     NaN  \n",
       "2                    Silent                     NaN                     NaN  \n",
       "3         Missense_Mutation                     NaN                     NaN  \n",
       "4                       RNA                     NaN                     NaN  \n",
       "...                     ...                     ...                     ...  \n",
       "3759                 Silent                     NaN                     NaN  \n",
       "3760                 Silent                     NaN                     NaN  \n",
       "3761      Missense_Mutation                     NaN                     NaN  \n",
       "3762      Missense_Mutation                     NaN                     NaN  \n",
       "3763      Nonsense_Mutation                     NaN                     NaN  \n",
       "\n",
       "[3764 rows x 12 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_data['chol'][['Hugo_Symbol', 'Entrez_Gene_Id', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'Reference_Allele', 'Variant_Type', 'Start_Position', 'End_Position', 'Chromosome', 'Variant_Classification', 'Match_Norm_Seq_Allele1',\n",
    "'Match_Norm_Seq_Allele2']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All clinical data with disease labels and mutation data saved as CSV files.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "# Directory containing the data files\n",
    "directory = '/Users/ethan/Downloads/tcga_data'\n",
    "\n",
    "# Reference index mapping cancer acronyms to project names\n",
    "disease_labels = {\n",
    "    \"sarc\": \"Sarcoma\",\n",
    "    \"kirp\": \"Kidney Renal Papillary Cell Carcinoma\",\n",
    "    \"esca\": \"Esophageal Carcinoma\",\n",
    "    \"hnsc\": \"Head and Neck Squamous Cell Carcinoma\",\n",
    "    \"chol\": \"Cholangiocarcinoma\",\n",
    "    \"prad\": \"Prostate Adenocarcinoma\",\n",
    "    \"ucec\": \"Uterine Corpus Endometrial Carcinoma\",\n",
    "    \"kich\": \"Kidney Chromophobe\",\n",
    "    \"laml\": \"Acute Myeloid Leukemia\",\n",
    "    \"ov\": \"Ovarian Serous Cystadenocarcinoma\",\n",
    "    \"uvm\": \"Uveal Melanoma\",\n",
    "    \"lgg\": \"Brain Lower Grade Glioma\",\n",
    "    \"brca\": \"Breast Invasive Carcinoma\",\n",
    "    \"lihc\": \"Liver Hepatocellular Carcinoma\",\n",
    "    \"luad\": \"Lung Adenocarcinoma\"\n",
    "}\n",
    "\n",
    "# Initialize dictionaries to store DataFrames for clinical and mutation files\n",
    "clinical_data = {}\n",
    "mutation_data = {}\n",
    "\n",
    "# Loop through each file in the directory\n",
    "for filename in os.listdir(directory):\n",
    "    cancer_type = filename.split('_')[0]\n",
    "    file_path = os.path.join(directory, filename)\n",
    "\n",
    "    try:\n",
    "        if filename.endswith('_clinical_patient.txt'):\n",
    "            # Read clinical patient file, skipping comment lines\n",
    "            df = pd.read_csv(file_path, sep='\\t', comment='#', header=0, engine='python')\n",
    "            # Add the Disease_Label column based on the mapping\n",
    "            df['Disease_Label'] = disease_labels.get(cancer_type, \"Unknown\")\n",
    "            clinical_data[cancer_type] = df\n",
    "        elif filename.endswith('_mutations.txt'):\n",
    "            if cancer_type == 'chol':\n",
    "                mutation_data[cancer_type] = pd.read_csv(file_path, sep='\\t', skiprows=2, engine='python')\n",
    "            else:\n",
    "                mutation_data[cancer_type] = pd.read_csv(file_path, sep='\\t', header=0, engine='python')\n",
    "    except pd.errors.ParserError as e:\n",
    "        print(f\"Error parsing {filename}: {e}\")\n",
    "\n",
    "# Combine all clinical DataFrames into one and save as CSV\n",
    "all_clinical_df = pd.concat(clinical_data.values(), ignore_index=True)\n",
    "all_clinical_df.to_csv('/Users/ethan/Downloads/all_clinical_data_with_labels.csv', index=False)\n",
    "\n",
    "# Combine all mutation DataFrames into one and save as CSV\n",
    "all_mutation_df = pd.concat(mutation_data.values(), ignore_index=True)\n",
    "all_mutation_df.to_csv('/Users/ethan/Downloads/all_mutation_data.csv', index=False)\n",
    "\n",
    "print(\"All clinical data with disease labels and mutation data saved as CSV files.\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>...</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>n_ref_count_val</th>\n",
       "      <th>tumor_vaf_val</th>\n",
       "      <th>n_alt_count_val</th>\n",
       "      <th>normal_vaf_val</th>\n",
       "      <th>dbSNP_Strength</th>\n",
       "      <th>Confidence</th>\n",
       "      <th>NVarRat</th>\n",
       "      <th>TVarRat</th>\n",
       "      <th>dbSNP_PopFreq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA10</td>\n",
       "      <td>10349.0</td>\n",
       "      <td>genome.wustl.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>67218779</td>\n",
       "      <td>67218780</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCC8</td>\n",
       "      <td>6833.0</td>\n",
       "      <td>genome.wustl.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>11</td>\n",
       "      <td>17418754</td>\n",
       "      <td>17418754</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACACA</td>\n",
       "      <td>31.0</td>\n",
       "      <td>genome.wustl.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>35518913</td>\n",
       "      <td>35518913</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ADAR</td>\n",
       "      <td>103.0</td>\n",
       "      <td>genome.wustl.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>154562755</td>\n",
       "      <td>154562755</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADAR</td>\n",
       "      <td>103.0</td>\n",
       "      <td>genome.wustl.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>154569329</td>\n",
       "      <td>154569329</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706834</th>\n",
       "      <td>TMEM170B</td>\n",
       "      <td>100113407.0</td>\n",
       "      <td>hgsc.bcm.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>6</td>\n",
       "      <td>11575790</td>\n",
       "      <td>11575797</td>\n",
       "      <td>+</td>\n",
       "      <td>coding_sequence_variant,3_prime_UTR_variant</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0</td>\n",
       "      <td>0.138889</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706835</th>\n",
       "      <td>CSMD2</td>\n",
       "      <td>114784.0</td>\n",
       "      <td>hgsc.bcm.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>34008360</td>\n",
       "      <td>34008360</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alt</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706836</th>\n",
       "      <td>CSMD2</td>\n",
       "      <td>114784.0</td>\n",
       "      <td>hgsc.bcm.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>33985175</td>\n",
       "      <td>33985175</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706837</th>\n",
       "      <td>CTDNEP1</td>\n",
       "      <td>23399.0</td>\n",
       "      <td>hgsc.bcm.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7147902</td>\n",
       "      <td>7147902</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706838</th>\n",
       "      <td>ALDH7A1</td>\n",
       "      <td>501.0</td>\n",
       "      <td>hgsc.bcm.edu</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>5</td>\n",
       "      <td>125930919</td>\n",
       "      <td>125930919</td>\n",
       "      <td>+</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>5'UTR</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>none</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>706839 rows × 348 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol  Entrez_Gene_Id            Center NCBI_Build Chromosome  \\\n",
       "0           ABCA10         10349.0  genome.wustl.edu     GRCh37         17   \n",
       "1            ABCC8          6833.0  genome.wustl.edu     GRCh37         11   \n",
       "2            ACACA            31.0  genome.wustl.edu     GRCh37         17   \n",
       "3             ADAR           103.0  genome.wustl.edu     GRCh37          1   \n",
       "4             ADAR           103.0  genome.wustl.edu     GRCh37          1   \n",
       "...            ...             ...               ...        ...        ...   \n",
       "706834    TMEM170B     100113407.0      hgsc.bcm.edu     GRCh37          6   \n",
       "706835       CSMD2        114784.0      hgsc.bcm.edu     GRCh37          1   \n",
       "706836       CSMD2        114784.0      hgsc.bcm.edu     GRCh37          1   \n",
       "706837     CTDNEP1         23399.0      hgsc.bcm.edu     GRCh37         17   \n",
       "706838     ALDH7A1           501.0      hgsc.bcm.edu     GRCh37          5   \n",
       "\n",
       "        Start_Position  End_Position Strand  \\\n",
       "0             67218779      67218780      +   \n",
       "1             17418754      17418754      +   \n",
       "2             35518913      35518913      +   \n",
       "3            154562755     154562755      +   \n",
       "4            154569329     154569329      +   \n",
       "...                ...           ...    ...   \n",
       "706834        11575790      11575797      +   \n",
       "706835        34008360      34008360      +   \n",
       "706836        33985175      33985175      +   \n",
       "706837         7147902       7147902      +   \n",
       "706838       125930919     125930919      +   \n",
       "\n",
       "                                        Consequence Variant_Classification  \\\n",
       "0                                frameshift_variant        Frame_Shift_Ins   \n",
       "1                                  missense_variant      Missense_Mutation   \n",
       "2                                  missense_variant      Missense_Mutation   \n",
       "3                                  missense_variant      Missense_Mutation   \n",
       "4                                  missense_variant      Missense_Mutation   \n",
       "...                                             ...                    ...   \n",
       "706834  coding_sequence_variant,3_prime_UTR_variant        Frame_Shift_Del   \n",
       "706835    synonymous_variant,NMD_transcript_variant                 Silent   \n",
       "706836    synonymous_variant,NMD_transcript_variant                 Silent   \n",
       "706837    synonymous_variant,NMD_transcript_variant                 Silent   \n",
       "706838                          5_prime_UTR_variant                  5'UTR   \n",
       "\n",
       "        ... dbSNP n_ref_count_val tumor_vaf_val n_alt_count_val  \\\n",
       "0       ...   NaN             NaN           NaN             NaN   \n",
       "1       ...   NaN             NaN           NaN             NaN   \n",
       "2       ...   NaN             NaN           NaN             NaN   \n",
       "3       ...   NaN             NaN           NaN             NaN   \n",
       "4       ...   NaN             NaN           NaN             NaN   \n",
       "...     ...   ...             ...           ...             ...   \n",
       "706834  ...   NaN             NaN           NaN             NaN   \n",
       "706835  ...   NaN             NaN           NaN             NaN   \n",
       "706836  ...   NaN             NaN           NaN             NaN   \n",
       "706837  ...   NaN             NaN           NaN             NaN   \n",
       "706838  ...   NaN             NaN           NaN             NaN   \n",
       "\n",
       "       normal_vaf_val dbSNP_Strength Confidence NVarRat   TVarRat  \\\n",
       "0                 NaN            NaN        NaN     NaN       NaN   \n",
       "1                 NaN            NaN        NaN     NaN       NaN   \n",
       "2                 NaN            NaN        NaN     NaN       NaN   \n",
       "3                 NaN            NaN        NaN     NaN       NaN   \n",
       "4                 NaN            NaN        NaN     NaN       NaN   \n",
       "...               ...            ...        ...     ...       ...   \n",
       "706834            NaN           none       PASS       0  0.138889   \n",
       "706835            NaN            alt       PASS       .         .   \n",
       "706836            NaN           none       PASS       .         .   \n",
       "706837            NaN           none       PASS       .         .   \n",
       "706838            NaN           none       PASS       .         .   \n",
       "\n",
       "       dbSNP_PopFreq  \n",
       "0                NaN  \n",
       "1                NaN  \n",
       "2                NaN  \n",
       "3                NaN  \n",
       "4                NaN  \n",
       "...              ...  \n",
       "706834             .  \n",
       "706835             .  \n",
       "706836             .  \n",
       "706837             .  \n",
       "706838             .  \n",
       "\n",
       "[706839 rows x 348 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_mutation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         TCGA-JY-A6FG-01\n",
       "1         TCGA-JY-A6FG-01\n",
       "2         TCGA-JY-A6FG-01\n",
       "3         TCGA-JY-A6FG-01\n",
       "4         TCGA-JY-A6FG-01\n",
       "               ...       \n",
       "706834    TCGA-5P-A9KC-01\n",
       "706835    TCGA-WN-AB4C-01\n",
       "706836    TCGA-A4-8630-01\n",
       "706837    TCGA-PJ-A5Z8-01\n",
       "706838    TCGA-BQ-5881-01\n",
       "Name: Tumor_Sample_Barcode, Length: 706839, dtype: object"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_mutation_df['Tumor_Sample_Barcode']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTHER_PATIENT_ID</th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>FORM_COMPLETION_DATE</th>\n",
       "      <th>PROSPECTIVE_COLLECTION</th>\n",
       "      <th>RETROSPECTIVE_COLLECTION</th>\n",
       "      <th>SEX</th>\n",
       "      <th>MENOPAUSE_STATUS</th>\n",
       "      <th>RACE</th>\n",
       "      <th>ETHNICITY</th>\n",
       "      <th>HISTORY_OTHER_MALIGNANCY</th>\n",
       "      <th>...</th>\n",
       "      <th>PRIMARY_DIAGNOSIS</th>\n",
       "      <th>YEAR_OF_DIAGNOSIS</th>\n",
       "      <th>BIOPSY_SITE</th>\n",
       "      <th>PATH_STAGE</th>\n",
       "      <th>MORPHOLOGY</th>\n",
       "      <th>PRIOR_TREATMENT</th>\n",
       "      <th>PRIOR_MALIGNANCY</th>\n",
       "      <th>PROJECT_STATE</th>\n",
       "      <th>SARCOMATOID_FEATURES</th>\n",
       "      <th>SARCOMATOID_PERCENT_OF_TUMOR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6E7D5EC6-A469-467C-B748-237353C23416</td>\n",
       "      <td>TCGA-3C-AAAU</td>\n",
       "      <td>1/13/14</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Female</td>\n",
       "      <td>Pre (&lt;6 months since LMP AND no prior bilatera...</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>55262FCB-1B01-4480-B322-36570430C917</td>\n",
       "      <td>TCGA-3C-AALI</td>\n",
       "      <td>7/28/14</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Female</td>\n",
       "      <td>Post (prior bilateral ovariectomy OR &gt;12 mo si...</td>\n",
       "      <td>BLACK OR AFRICAN AMERICAN</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>427D0648-3F77-4FFC-B52C-89855426D647</td>\n",
       "      <td>TCGA-3C-AALJ</td>\n",
       "      <td>7/28/14</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Female</td>\n",
       "      <td>Post (prior bilateral ovariectomy OR &gt;12 mo si...</td>\n",
       "      <td>BLACK OR AFRICAN AMERICAN</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C31900A4-5DCD-4022-97AC-638E86E889E4</td>\n",
       "      <td>TCGA-3C-AALK</td>\n",
       "      <td>7/28/14</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Female</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>BLACK OR AFRICAN AMERICAN</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6623FC5E-00BE-4476-967A-CBD55F676EA6</td>\n",
       "      <td>TCGA-4H-AAAK</td>\n",
       "      <td>11/13/14</td>\n",
       "      <td>YES</td>\n",
       "      <td>NO</td>\n",
       "      <td>Female</td>\n",
       "      <td>Post (prior bilateral ovariectomy OR &gt;12 mo si...</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5850</th>\n",
       "      <td>DFBE9B61-1F13-4BD1-8F87-364B186C69B5</td>\n",
       "      <td>TCGA-UW-A7GP</td>\n",
       "      <td>12/17/13</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5851</th>\n",
       "      <td>A86C2490-37DA-46A3-88E2-0B0CE21D067F</td>\n",
       "      <td>TCGA-UW-A7GR</td>\n",
       "      <td>12/17/13</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>Yes, History of Prior Malignancy</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5852</th>\n",
       "      <td>F300EFC6-A125-468A-BB2F-FBAAC71FCCA0</td>\n",
       "      <td>TCGA-UW-A7GU</td>\n",
       "      <td>12/17/13</td>\n",
       "      <td>NO</td>\n",
       "      <td>YES</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BLACK OR AFRICAN AMERICAN</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5853</th>\n",
       "      <td>D0A8CFDC-C025-4DE9-94AD-F5160809FF8F</td>\n",
       "      <td>TCGA-UW-A7GX</td>\n",
       "      <td>12/17/13</td>\n",
       "      <td>YES</td>\n",
       "      <td>NO</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BLACK OR AFRICAN AMERICAN</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5854</th>\n",
       "      <td>E0901BB2-27CE-46A4-BD82-EE1A8EBBE48C</td>\n",
       "      <td>TCGA-UW-A7GY</td>\n",
       "      <td>12/17/13</td>\n",
       "      <td>YES</td>\n",
       "      <td>NO</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>NOT HISPANIC OR LATINO</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5855 rows × 364 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          OTHER_PATIENT_ID    PATIENT_ID FORM_COMPLETION_DATE  \\\n",
       "0     6E7D5EC6-A469-467C-B748-237353C23416  TCGA-3C-AAAU              1/13/14   \n",
       "1     55262FCB-1B01-4480-B322-36570430C917  TCGA-3C-AALI              7/28/14   \n",
       "2     427D0648-3F77-4FFC-B52C-89855426D647  TCGA-3C-AALJ              7/28/14   \n",
       "3     C31900A4-5DCD-4022-97AC-638E86E889E4  TCGA-3C-AALK              7/28/14   \n",
       "4     6623FC5E-00BE-4476-967A-CBD55F676EA6  TCGA-4H-AAAK             11/13/14   \n",
       "...                                    ...           ...                  ...   \n",
       "5850  DFBE9B61-1F13-4BD1-8F87-364B186C69B5  TCGA-UW-A7GP             12/17/13   \n",
       "5851  A86C2490-37DA-46A3-88E2-0B0CE21D067F  TCGA-UW-A7GR             12/17/13   \n",
       "5852  F300EFC6-A125-468A-BB2F-FBAAC71FCCA0  TCGA-UW-A7GU             12/17/13   \n",
       "5853  D0A8CFDC-C025-4DE9-94AD-F5160809FF8F  TCGA-UW-A7GX             12/17/13   \n",
       "5854  E0901BB2-27CE-46A4-BD82-EE1A8EBBE48C  TCGA-UW-A7GY             12/17/13   \n",
       "\n",
       "     PROSPECTIVE_COLLECTION RETROSPECTIVE_COLLECTION     SEX  \\\n",
       "0                        NO                      YES  Female   \n",
       "1                        NO                      YES  Female   \n",
       "2                        NO                      YES  Female   \n",
       "3                        NO                      YES  Female   \n",
       "4                       YES                       NO  Female   \n",
       "...                     ...                      ...     ...   \n",
       "5850                     NO                      YES    Male   \n",
       "5851                     NO                      YES  Female   \n",
       "5852                     NO                      YES    Male   \n",
       "5853                    YES                       NO    Male   \n",
       "5854                    YES                       NO  Female   \n",
       "\n",
       "                                       MENOPAUSE_STATUS  \\\n",
       "0     Pre (<6 months since LMP AND no prior bilatera...   \n",
       "1     Post (prior bilateral ovariectomy OR >12 mo si...   \n",
       "2     Post (prior bilateral ovariectomy OR >12 mo si...   \n",
       "3                                       [Not Available]   \n",
       "4     Post (prior bilateral ovariectomy OR >12 mo si...   \n",
       "...                                                 ...   \n",
       "5850                                                NaN   \n",
       "5851                                                NaN   \n",
       "5852                                                NaN   \n",
       "5853                                                NaN   \n",
       "5854                                                NaN   \n",
       "\n",
       "                           RACE               ETHNICITY  \\\n",
       "0                         WHITE  NOT HISPANIC OR LATINO   \n",
       "1     BLACK OR AFRICAN AMERICAN  NOT HISPANIC OR LATINO   \n",
       "2     BLACK OR AFRICAN AMERICAN  NOT HISPANIC OR LATINO   \n",
       "3     BLACK OR AFRICAN AMERICAN  NOT HISPANIC OR LATINO   \n",
       "4                         WHITE  NOT HISPANIC OR LATINO   \n",
       "...                         ...                     ...   \n",
       "5850                      WHITE         [Not Available]   \n",
       "5851                      WHITE         [Not Available]   \n",
       "5852  BLACK OR AFRICAN AMERICAN         [Not Available]   \n",
       "5853  BLACK OR AFRICAN AMERICAN  NOT HISPANIC OR LATINO   \n",
       "5854                      WHITE  NOT HISPANIC OR LATINO   \n",
       "\n",
       "              HISTORY_OTHER_MALIGNANCY  ... PRIMARY_DIAGNOSIS  \\\n",
       "0                                   No  ...               NaN   \n",
       "1                                   No  ...               NaN   \n",
       "2                                   No  ...               NaN   \n",
       "3                                   No  ...               NaN   \n",
       "4                                   No  ...               NaN   \n",
       "...                                ...  ...               ...   \n",
       "5850                                No  ...               NaN   \n",
       "5851  Yes, History of Prior Malignancy  ...               NaN   \n",
       "5852                                No  ...               NaN   \n",
       "5853                                No  ...               NaN   \n",
       "5854                                No  ...               NaN   \n",
       "\n",
       "     YEAR_OF_DIAGNOSIS BIOPSY_SITE PATH_STAGE MORPHOLOGY PRIOR_TREATMENT  \\\n",
       "0                  NaN         NaN        NaN        NaN             NaN   \n",
       "1                  NaN         NaN        NaN        NaN             NaN   \n",
       "2                  NaN         NaN        NaN        NaN             NaN   \n",
       "3                  NaN         NaN        NaN        NaN             NaN   \n",
       "4                  NaN         NaN        NaN        NaN             NaN   \n",
       "...                ...         ...        ...        ...             ...   \n",
       "5850               NaN         NaN        NaN        NaN             NaN   \n",
       "5851               NaN         NaN        NaN        NaN             NaN   \n",
       "5852               NaN         NaN        NaN        NaN             NaN   \n",
       "5853               NaN         NaN        NaN        NaN             NaN   \n",
       "5854               NaN         NaN        NaN        NaN             NaN   \n",
       "\n",
       "     PRIOR_MALIGNANCY PROJECT_STATE SARCOMATOID_FEATURES  \\\n",
       "0                 NaN           NaN                  NaN   \n",
       "1                 NaN           NaN                  NaN   \n",
       "2                 NaN           NaN                  NaN   \n",
       "3                 NaN           NaN                  NaN   \n",
       "4                 NaN           NaN                  NaN   \n",
       "...               ...           ...                  ...   \n",
       "5850              NaN           NaN      [Not Available]   \n",
       "5851              NaN           NaN      [Not Available]   \n",
       "5852              NaN           NaN      [Not Available]   \n",
       "5853              NaN           NaN                   NO   \n",
       "5854              NaN           NaN                   NO   \n",
       "\n",
       "     SARCOMATOID_PERCENT_OF_TUMOR  \n",
       "0                             NaN  \n",
       "1                             NaN  \n",
       "2                             NaN  \n",
       "3                             NaN  \n",
       "4                             NaN  \n",
       "...                           ...  \n",
       "5850              [Not Available]  \n",
       "5851              [Not Available]  \n",
       "5852              [Not Available]  \n",
       "5853              [Not Available]  \n",
       "5854              [Not Available]  \n",
       "\n",
       "[5855 rows x 364 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/pj/894szprx04q1zqpqhy4bgvqr0000gn/T/ipykernel_64386/1745692416.py:4: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  all_mutation_df['PATIENT_ID'] = all_mutation_df['Tumor_Sample_Barcode'].str[:12]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged dataset saved as 'merged_clinical_mutation_data.csv'.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Extract patient identifier (first 12 characters) from Tumor_Sample_Barcode in mutation data\n",
    "all_mutation_df['PATIENT_ID'] = all_mutation_df['Tumor_Sample_Barcode'].str[:12]\n",
    "\n",
    "# Specify the columns to keep from mutation data\n",
    "mutation_columns = [\n",
    "    \"Hugo_Symbol\", \"Entrez_Gene_Id\", \"Chromosome\", \"Start_Position\",\n",
    "    \"End_Position\", \"Strand\", \"Consequence\", \"Variant_Classification\",\n",
    "    \"Variant_Type\", \"Reference_Allele\", \"Tumor_Seq_Allele1\",\n",
    "    \"Tumor_Seq_Allele2\", \"Mutation_Status\", \"t_ref_count\", \"t_alt_count\",\n",
    "    \"n_ref_count\", \"n_alt_count\", \"PATIENT_ID\"\n",
    "]\n",
    "\n",
    "# Specify the columns to keep from clinical data\n",
    "clinical_columns = [\"PATIENT_ID\", \"RACE\", \"SEX\", 'Disease_Label']\n",
    "\n",
    "# Filter the datasets to keep only the specified columns\n",
    "all_mutation_df = all_mutation_df[mutation_columns]\n",
    "all_clinical_df = all_clinical_df[clinical_columns]\n",
    "\n",
    "# Perform the merge (one-to-many join) on the PATIENT_ID column\n",
    "merged_df = pd.merge(all_mutation_df, all_clinical_df, on=\"PATIENT_ID\", how=\"inner\")\n",
    "\n",
    "# Save the resulting merged dataset to a CSV file\n",
    "merged_df.to_csv('/Users/ethan/Downloads/merged_clinical_mutation_data.csv', index=False)\n",
    "\n",
    "print(\"Merged dataset saved as 'merged_clinical_mutation_data.csv'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Column 'Hugo_Symbol': 0 NA/null values\n",
      "Column 'Entrez_Gene_Id': 15970 NA/null values\n",
      "Column 'Chromosome': 0 NA/null values\n",
      "Column 'Start_Position': 0 NA/null values\n",
      "Column 'End_Position': 0 NA/null values\n",
      "Column 'Strand': 42 NA/null values\n",
      "Column 'Consequence': 118 NA/null values\n",
      "Column 'Variant_Classification': 0 NA/null values\n",
      "Column 'Variant_Type': 0 NA/null values\n",
      "Column 'Reference_Allele': 0 NA/null values\n",
      "Column 'Tumor_Seq_Allele1': 0 NA/null values\n",
      "Column 'Tumor_Seq_Allele2': 0 NA/null values\n",
      "Column 'Mutation_Status': 0 NA/null values\n",
      "Column 't_ref_count': 103680 NA/null values\n",
      "Column 't_alt_count': 103680 NA/null values\n",
      "Column 'n_ref_count': 441493 NA/null values\n",
      "Column 'n_alt_count': 441493 NA/null values\n",
      "Column 'PATIENT_ID': 0 NA/null values\n",
      "Column 'RACE': 218 NA/null values\n",
      "Column 'SEX': 175 NA/null values\n",
      "Column 'Disease_Label': 0 NA/null values\n"
     ]
    }
   ],
   "source": [
    "na_counts = merged_df.isnull().sum()\n",
    "\n",
    "# Display the results\n",
    "for column, count in na_counts.items():\n",
    "    print(f\"Column '{column}': {count} NA/null values\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cleaned dataset saved as 'cleaned_BERT_gene_data.csv'.\n"
     ]
    }
   ],
   "source": [
    "# Replace NA values with empty strings in specific columns\n",
    "columns_to_fill = ['t_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count']\n",
    "merged_df[columns_to_fill] = merged_df[columns_to_fill].fillna('')\n",
    "\n",
    "# Drop rows with NA values in the remaining columns\n",
    "cleaned_df = merged_df.dropna()\n",
    "\n",
    "# Save the cleaned dataset to a CSV file\n",
    "cleaned_df.to_csv('/Users/ethan/Downloads/cleaned_BERT_gene_data.csv', index=False)\n",
    "\n",
    "print(\"Cleaned dataset saved as 'cleaned_BERT_gene_data.csv'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Uterine Corpus Endometrial Carcinoma     182949\n",
       "Head and Neck Squamous Cell Carcinoma    119607\n",
       "Breast Invasive Carcinoma                 86641\n",
       "Lung Adenocarcinoma                       67939\n",
       "Esophageal Carcinoma                      56961\n",
       "Liver Hepatocellular Carcinoma            53722\n",
       "Prostate Adenocarcinoma                   39740\n",
       "Ovarian Serous Cystadenocarcinoma         20057\n",
       "Kidney Renal Papillary Cell Carcinoma     18863\n",
       "Sarcoma                                   18533\n",
       "Brain Lower Grade Glioma                   9876\n",
       "Kidney Chromophobe                         7543\n",
       "Cholangiocarcinoma                         3546\n",
       "Acute Myeloid Leukemia                     2475\n",
       "Uveal Melanoma                             2156\n",
       "Name: Disease_Label, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df['Disease_Label'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Hugo_Symbol', 'Entrez_Gene_Id', 'Chromosome', 'Start_Position',\n",
       "       'End_Position', 'Strand', 'Consequence', 'Variant_Classification',\n",
       "       'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1',\n",
       "       'Tumor_Seq_Allele2', 'Mutation_Status', 't_ref_count', 't_alt_count',\n",
       "       'n_ref_count', 'n_alt_count', 'PATIENT_ID', 'RACE', 'SEX',\n",
       "       'Disease_Label'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/pj/894szprx04q1zqpqhy4bgvqr0000gn/T/ipykernel_64386/1865712646.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cleaned_df['Combined'] = cleaned_df[columns_to_concatenate].astype(str).agg(' '.join, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed patient-level dataset saved as 'patient_level_data.csv'.\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to concatenate\n",
    "columns_to_concatenate = [\n",
    "    'Hugo_Symbol', 'Entrez_Gene_Id', 'Chromosome', 'Start_Position',\n",
    "    'End_Position', 'Strand', 'Consequence', 'Variant_Classification',\n",
    "    'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1',\n",
    "    'Tumor_Seq_Allele2', 'Mutation_Status', 't_ref_count', 't_alt_count',\n",
    "    'n_ref_count', 'n_alt_count', 'RACE', 'SEX'\n",
    "]\n",
    "\n",
    "# Concatenate specified columns into one field for each row\n",
    "cleaned_df['Combined'] = cleaned_df[columns_to_concatenate].astype(str).agg(' '.join, axis=1)\n",
    "\n",
    "# Group by PATIENT_ID and aggregate rows into one by concatenating 'Combined' values\n",
    "patient_level_df = (\n",
    "    cleaned_df.groupby(['PATIENT_ID'], as_index=False)\n",
    "    .agg({'Combined': ' '.join, 'Disease_Label': 'first'})  # Use 'first' to keep the same Disease_Label for the patient\n",
    ")\n",
    "\n",
    "# Set PATIENT_ID as the index\n",
    "patient_level_df = patient_level_df.set_index('PATIENT_ID')\n",
    "\n",
    "# Save the resulting dataset\n",
    "patient_level_df.to_csv('/Users/ethan/Downloads/patient_level_data.csv')\n",
    "\n",
    "print(\"Processed patient-level dataset saved as 'patient_level_data.csv'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Combined</th>\n",
       "      <th>Disease_Label</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-04-1331</th>\n",
       "      <td>BAI2 576.0 1 32202221 32202221 + missense_vari...</td>\n",
       "      <td>Ovarian Serous Cystadenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-04-1332</th>\n",
       "      <td>SERPINF1 5176.0 17 1674326 1674326 + missense_...</td>\n",
       "      <td>Ovarian Serous Cystadenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-04-1336</th>\n",
       "      <td>CPXM1 56265.0 20 2778850 2778850 + missense_va...</td>\n",
       "      <td>Ovarian Serous Cystadenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-04-1337</th>\n",
       "      <td>ABCA3 21.0 16 2336768 2336768 + missense_varia...</td>\n",
       "      <td>Ovarian Serous Cystadenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-04-1338</th>\n",
       "      <td>A2M 2.0 12 9230420 9230420 + missense_variant ...</td>\n",
       "      <td>Ovarian Serous Cystadenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZS-A9CD</th>\n",
       "      <td>ARID1A 8289.0 1 27106649 27106649 + missense_v...</td>\n",
       "      <td>Liver Hepatocellular Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZS-A9CE</th>\n",
       "      <td>TCEA3 6920.0 1 23720511 23720511 + missense_va...</td>\n",
       "      <td>Liver Hepatocellular Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZS-A9CF</th>\n",
       "      <td>PTCHD2 57540.0 1 11596435 11596435 + missense_...</td>\n",
       "      <td>Liver Hepatocellular Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZS-A9CG</th>\n",
       "      <td>HSPG2 3339.0 1 22178363 22178363 + synonymous_...</td>\n",
       "      <td>Liver Hepatocellular Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZU-A8S4</th>\n",
       "      <td>ARID1A 8289.0 1 26779357 26779361 + frameshift...</td>\n",
       "      <td>Cholangiocarcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4542 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Combined  \\\n",
       "PATIENT_ID                                                        \n",
       "TCGA-04-1331  BAI2 576.0 1 32202221 32202221 + missense_vari...   \n",
       "TCGA-04-1332  SERPINF1 5176.0 17 1674326 1674326 + missense_...   \n",
       "TCGA-04-1336  CPXM1 56265.0 20 2778850 2778850 + missense_va...   \n",
       "TCGA-04-1337  ABCA3 21.0 16 2336768 2336768 + missense_varia...   \n",
       "TCGA-04-1338  A2M 2.0 12 9230420 9230420 + missense_variant ...   \n",
       "...                                                         ...   \n",
       "TCGA-ZS-A9CD  ARID1A 8289.0 1 27106649 27106649 + missense_v...   \n",
       "TCGA-ZS-A9CE  TCEA3 6920.0 1 23720511 23720511 + missense_va...   \n",
       "TCGA-ZS-A9CF  PTCHD2 57540.0 1 11596435 11596435 + missense_...   \n",
       "TCGA-ZS-A9CG  HSPG2 3339.0 1 22178363 22178363 + synonymous_...   \n",
       "TCGA-ZU-A8S4  ARID1A 8289.0 1 26779357 26779361 + frameshift...   \n",
       "\n",
       "                                  Disease_Label  \n",
       "PATIENT_ID                                       \n",
       "TCGA-04-1331  Ovarian Serous Cystadenocarcinoma  \n",
       "TCGA-04-1332  Ovarian Serous Cystadenocarcinoma  \n",
       "TCGA-04-1336  Ovarian Serous Cystadenocarcinoma  \n",
       "TCGA-04-1337  Ovarian Serous Cystadenocarcinoma  \n",
       "TCGA-04-1338  Ovarian Serous Cystadenocarcinoma  \n",
       "...                                         ...  \n",
       "TCGA-ZS-A9CD     Liver Hepatocellular Carcinoma  \n",
       "TCGA-ZS-A9CE     Liver Hepatocellular Carcinoma  \n",
       "TCGA-ZS-A9CF     Liver Hepatocellular Carcinoma  \n",
       "TCGA-ZS-A9CG     Liver Hepatocellular Carcinoma  \n",
       "TCGA-ZU-A8S4                 Cholangiocarcinoma  \n",
       "\n",
       "[4542 rows x 2 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_level_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The maximum length of the values in the 'Combined' column is: 17888.33025099075\n"
     ]
    }
   ],
   "source": [
    "# Calculate the length of each value in the 'Combined' column\n",
    "combined_lengths = patient_level_df['Combined'].str.len()\n",
    "\n",
    "# Find the maximum length\n",
    "max_length = combined_lengths.mean()\n",
    "\n",
    "print(f\"The maximum length of the values in the 'Combined' column is: {max_length}\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.6 ('ethan': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "cd13eb32261735dadd329c5dd164f174ef8ee121db3ad7b71e739d507bc142dd"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
